bsp/esp32_p4_function_ev_board/include/bsp/esp32_p4.h

```diff
@@ -0,0 +1,322 @@
+ *      - ESP_OK on success
+ *      - ESP_ERR_NOT_FOUND if the partition table does not contain SPIFFS partition with given label
+ *      - ESP_ERR_INVALID_STATE if esp_vfs_spiffs_unregister was already called
+ *      - ESP_ERR_NO_MEM if memory can not be allocated
+ *      - ESP_FAIL if partition can not be mounted
+ *      - other error codes
+ */
+esp_err_t bsp_spiffs_unmount(void);
+
+/**************************************************************************************************
+ *
+ * uSD card
+ *
+ * After mounting the uSD card, it can be accessed with stdio functions ie.:
+ * \code{.c}
+ * FILE* f = fopen(BSP_MOUNT_POINT"/hello.txt", "w");
+ * fprintf(f, "Hello %s!\n", bsp_sdcard->cid.name);
+ * fclose(f);
+ * \endcode
+ **************************************************************************************************/
+#define BSP_SD_MOUNT_POINT      CONFIG_BSP_SD_MOUNT_POINT
+extern sdmmc_card_t *bsp_sdcard;
+
+/**
+ * @brief Mount microSD card to virtual file system
+ *
+ * @return
+ *      - ESP_OK on success
+ *      - ESP_ERR_INVALID_STATE if esp_vfs_fat_sdmmc_mount was already called
+ *      - ESP_ERR_NO_MEM if memory cannot be allocated
+ *      - ESP_FAIL if partition cannot be mounted
+ *      - other error codes from SDMMC or SPI drivers, SDMMC protocol, or FATFS drivers
+ */
+esp_err_t bsp_sdcard_mount(void);
+
+/**
+ * @brief Unmount microSD card from virtual file system
+ *
+ * @return
+ *      - ESP_OK on success
+ *      - ESP_ERR_NOT_FOUND if the partition table does not contain FATFS partition with given label
+ *      - ESP_ERR_INVALID_STATE if esp_vfs_fat_spiflash_mount was already called
+ *      - ESP_ERR_NO_MEM if memory can not be allocated
+ *      - ESP_FAIL if partition can not be mounted
+ *      - other error codes from wear levelling library, SPI flash driver, or FATFS drivers
+ */
+esp_err_t bsp_sdcard_unmount(void);
+
+/**************************************************************************************************
+ *
+ * LCD interface
+ *
+ * ESP-BOX is shipped with 2.4inch ST7789 display controller.
+ * It features 16-bit colors, 320x240 resolution and capacitive touch controller.
+ *
+ * LVGL is used as graphics library. LVGL is NOT thread safe, therefore the user must take LVGL mutex
+ * by calling bsp_display_lock() before calling and LVGL API (lv_...) and then give the mutex with
+ * bsp_display_unlock().
+ *
+ * Display's backlight must be enabled explicitly by calling bsp_display_backlight_on()
+ **************************************************************************************************/
+#define BSP_LCD_H_RES              (1280)
+#define BSP_LCD_V_RES              (800)
+#define BSP_LCD_PIXEL_CLOCK_MHZ     (80)
+
+#define BSP_LCD_MIPI_DSI_LCD_HSYNC    (40)
+#define BSP_LCD_MIPI_DSI_LCD_HBP      (140)
+#define BSP_LCD_MIPI_DSI_LCD_HFP      (40)
+#define BSP_LCD_MIPI_DSI_LCD_VSYNC    (4)
+#define BSP_LCD_MIPI_DSI_LCD_VBP      (16)
+#define BSP_LCD_MIPI_DSI_LCD_VFP      (16)
```

Comment:
Could we move these definitions to display.h?

the idea here is, that in case the user wants to use _only_ the LCD part of a BSP, he must only include bsp/display.h. (not the full bsp/esp32_p4.h)